scripts/windows/test/Run-E2ETest.ps1

```diff
@@ -620,26 +654,152 @@ Function RunTempSensorTest
+            }
+            Write-Host "Run LeafDevice with X.509 CA auth in scope"
+            New-CACertsDevice "$leafDeviceId"
+            $testCommand = "&$LeafDeviceExeTestPath ``
+                -d `"$leafDeviceId`" ``
+                -c `"$IoTHubConnectionString`" ``
+                -e `"$EventHubConnectionString`" ``
+                -proto `"$protocol`" ``
+                -ct `"$TrustedCACertificatePath`" ``
+                -cac `"$EdgeCertGenScriptDir\certs\iot-device-${leafDeviceId}-full-chain.cert.pem`" ``
+                -cak `"$EdgeCertGenScriptDir\private\iot-device-${leafDeviceId}.key.pem`" ``
+                -ed-id `"$edgeDeviceId`" ``
+                -ed `"$env:computername`""
+            break
+        }
+
+        "x509Thumprint"
+        {
+            if ([string]::IsNullOrWhiteSpace($edgeDeviceId))
+            {
+                $(Throw "For X.509 leaf device, the Edge device Id is requried")
+            }
+            Write-Host "Run LeafDevice with X.509 thumbprint auth in scope"
+            New-CACertsDevice "$leafDeviceId-pri"
+            New-CACertsDevice "$leafDeviceId-sec"
+            $testCommand = "&$LeafDeviceExeTestPath ``
+                -d `"$leafDeviceId`" ``
+                -c `"$IoTHubConnectionString`" ``
+                -e `"$EventHubConnectionString`" ``
+                -proto `"$protocol`" ``
+                -ct `"$TrustedCACertificatePath`" ``
+                -ctpc `"$EdgeCertGenScriptDir\certs\iot-device-${leafDeviceId}-pri-full-chain.cert.pem`" ``
+                -ctpk `"$EdgeCertGenScriptDir\private\iot-device-${leafDeviceId}-pri.key.pem`" ``
+                -ctsc `"$EdgeCertGenScriptDir\certs\iot-device-${leafDeviceId}-sec-full-chain.cert.pem`" ``
+                -ctsk `"$EdgeCertGenScriptDir\private\iot-device-${leafDeviceId}-sec.key.pem`" ``
+                -ed-id `"$edgeDeviceId`" ``
+                -ed `"$env:computername`""
+            break
+        }
+
+        default
+        {
+            $(Throw "Unsupported auth mode $authType")
+        }
+     }
+
+    $testStartAt = Get-Date
+    Invoke-Expression $testCommand | Out-Host
+    $testExitCode = $LastExitCode
+    PrintLogs $testStartAt $testExitCode
+
+    Return $testExitCode
+}
+
 Function RunTransparentGatewayTest
 {
     PrintHighlightedMessage "Run Transparent Gateway test for $Architecture"
+
+    if ([string]::IsNullOrWhiteSpace($EdgeE2ERootCACertRSAFile))
+    {
+        $EdgeE2ERootCACertRSAFile=$DefaultInstalledRSARootCACert
+    }
+    if ([string]::IsNullOrWhiteSpace($EdgeE2ERootCAKeyRSAFile))
+    {
+        $EdgeE2ERootCAKeyRSAFile=$DefaultInstalledRSARootCAKey
+    }
     TestSetup
 
     $testStartAt = Get-Date
     $deviceId = "e2e-${ReleaseLabel}-Windows-${Architecture}-TransGW"
     PrintHighlightedMessage "Run quickstart test with -d ""$deviceId"" started at $testStartAt."
 
+    # setup environment before invoking cert gen script
```

Comment:
Please warp line772-789 in a single method.

Reply:
Thanks food suggestion. Fixed.